src/fixture.js

```diff
@@ -14,3 +15,8 @@ export const test = base.extend({
         await use(new ShopingCartPage(page));
     },
 });
+
+test.beforeEach(async ({ loginPage }) => {
```

Comment:
I'd rather not having before each in fixture as it will make every single test perform login by one user. 
This means you won't be bale to test negative scenarios or use other users for login

Reply:
So you mean it's unnecessary to put it in a fixture, but to indicate it in each test?
```
test('Sorting by name (A to Z)', async ({ inventoryPage, loginPage }) => {
    await loginPage.navigate();
    await loginPage.performLogin(username, password);
```

Reply:
Yes, it will give more flexibility with user